x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -275,27 +297,50 @@ void deleteSnapshots(Map<String, List<SnapshotInfo>> snapshotsToDelete,
 
     /**
      * Delete the given snapshot from the repository in blocking manner
+     *
+     * @param repo         The repository the snapshot is in
+     * @param snapshot     The snapshot metadata
+     * @param onCompletion A callback taking info on the history of the snapshot. If present, a SnapshotHistoryItem containing the results
+     *                     of the deletion. Empty if interrupted or failed to serialize exception.
      */
-    void deleteSnapshot(String slmPolicy, String repo, SnapshotId snapshot, SnapshotLifecycleStats slmStats) {
-        logger.info("[{}] snapshot retention deleting snapshot [{}]", repo, snapshot);
+    void deleteSnapshot(String slmPolicy, String repo, SnapshotInfo snapshot, SnapshotLifecycleStats slmStats,
+                        Consumer<Optional<SnapshotHistoryItem>> onCompletion) {
```

Comment:
I think it makes more sense for this to be `Consumer<SnapshotHistoryItem>` and the `Optional` wrapping happen inside of the function itself, generally Optional isn't a parameter that's supposed to be passed around between functions